src/code/views/node-view.coffee

```diff
@@ -17,50 +17,65 @@ NodeTitle = React.createFactory React.createClass
   mixins: [require '../mixins/node-title']
 
   getInitialState: ->
-    isUniqueTitle: @isUniqueTitle @props.node.title
+    @getStateFromProps @props
 
-  isUniqueTitle: (title) ->
-    @props.graphStore.isUniqueTitle title, @props.node
+  isUniqueTitle: (title, props) ->
+    @props.graphStore.isUniqueTitle title, (props or @props).node
 
   componentWillUnmount: ->
     if @props.isEditing
       @inputElm().off()
 
   componentWillUpdate: (nextProps) ->
-    # mark the title as updated even if no change was made when it leaves edit mode
-    @titleUpdated = true if @props.isEditing and not nextProps.isEditing
+    if @props.isEditing and not nextProps.isEditing
+      # mark the title as updated even if no change was made when it leaves edit mode
+      @titleUpdated = true
+    else if not @props.isEditing and nextProps.isEditing
```

Comment:
This one line was a little tricky to wrap my head around, but otherwise this whole PR is good.

Thanks!